packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -182,6 +185,34 @@ export default util.createRule<Options, MessageIds>({
       );
     }
 
+    function isAsyncMemberOrProperty(
+      propertyOrMemberNode:
+        | TSESTree.PropertyNonComputedName
+        | TSESTree.TSMethodSignatureNonComputedName
+        | TSESTree.PropertyDefinitionNonComputedName
+        | TSESTree.TSAbstractPropertyDefinitionNonComputedName
+        | TSESTree.MethodDefinitionNonComputedName
+        | TSESTree.TSAbstractMethodDefinitionNonComputedName,
+    ): boolean {
+      return Boolean(
+        'value' in propertyOrMemberNode &&
```

Comment:
Looking at this again: we generally try not to use this string `in` checking pattern, as it's a little imprecise. We normally check for specific types. But I can see why you did it here (much less code!), and there's already an instance of it in the file.

I think it's fine to check in as-is, and we should separately look into making a lint rule internally that flags this kind of check.

I also think I'd meant to post this in the first review but forgot 😄

Reply:
> as it's a little imprecise

I agree, and its not great for readability also, its not really clear what the intent of the `in` is, in terms of what node types its trying to check for, but I went with it since it was used everywhere in that file

> I can see why you did it here (much less code!)
Yeah I don't do it like this usually, but I was also surprised how short the code gets, but I'm still a fan of the type utils for better communication (but please don't ask me change it now😅)

Reply:
Haha don't worry, I won't 😄

Reply:
https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/326